dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/servlet/DisEnvCompatibilityCheck.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2004-2022, University of Oslo
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.webapi.servlet;
+
+import javax.servlet.ServletContext;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.springframework.core.annotation.Order;
+import org.springframework.web.WebApplicationInitializer;
+
+/**
+ * Checks environment compatibility for DHIS2 at startup.
+ *
+ * @author Jan Bernitt
+ */
+@Slf4j
+@Order( 1 )
+public class DisEnvCompatibilityCheck implements WebApplicationInitializer
+{
+    @Override
+    public void onStartup( ServletContext context )
+    {
+        String javaVersion = System.getProperty( "java.version" );
+        log.info( "Java Version: " + javaVersion );
+        if ( javaVersion.startsWith( "1." )
+            || javaVersion.startsWith( "9" )
+            || javaVersion.startsWith( "10" ) )
+        {
+            log.error( "DHIS2 requires Java 11+" );
```

Comment:
Perhaps add something like "Aborting DHIS 2 startup process, Java JDK 11+ is required" so that local sysamdmins understand what's going on?

Reply:
updated now.